sdk/appcenter-data/src/test/java/com/microsoft/appcenter/data/DataTest.java

```diff
@@ -1374,11 +1419,131 @@ public void shouldNotOperateOnLocalStorageWhenPartitionNotExist() {
         assertTrue(deleteDocument.getError().getMessage().contains(failedMessage));
     }
 
+    @Test
+    public void listUserPartitionReturnsAnExceptionWhenNotSignedIn() {
+        mockSignOut();
+        PaginatedDocuments<TestDocument> documents = Data.list(TestDocument.class, USER_DOCUMENTS).get();
+        Iterator<DocumentWrapper<TestDocument>> iterator = documents.iterator();
+        assertFalse(iterator.hasNext());
+        assertNotNull(documents);
+        Page<TestDocument> currentPage = documents.getCurrentPage();
+        assertNotNull(currentPage);
+        DataException error = currentPage.getError();
+        assertNull(currentPage.getItems());
+        assertNotNull(error);
+        assertTrue(error.getMessage().contains("List operation requested on user partition, but the user is not logged in."));
+    }
+
+    private void mockSignOut() {
+        Mockito.when(mAuthTokenContext.getAccountId()).thenReturn(null);
+    }
+
+    @Test
+    public void listAnObjectWhenThereArePendingOperations() {
+        listAnObjectWhenThereArePendingOperations(Utils.getGson().toJson(new TestDocument("test")), TestDocument.class);
+    }
+
+    @Test
+    public void listPrimitiveTypeWhenThereArePendingOperations() {
+        listAnObjectWhenThereArePendingOperations("document", String.class);
+    }
+
+    private <T> void listAnObjectWhenThereArePendingOperations(String document, Class<T> documentType) {
```

Comment:
This test should test with multiple documents

Reply:
As discussed offline there will be a PR to improve tests.